### PR TITLE
Create integration test of ReferenceModeStore and DatabaseDriver

### PR DESCRIPTION
**Pull request by jasonwyatt**
Makes sure the DatabaseDriver can play nicely with the ReferenceModeStore.

### CODE REVIEW

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+                Reference("an-id", activeStore.backingStore.storageKey, VersionMap(actor to 1))
+            )
+
+        val bobKey = activeStore.backingStore.storageKey.childKeyWithComponent("an-id")
+        val capturedBob = requireNotNull(
+            database.get(bobKey, DatabaseData.Entity::class) as? DatabaseData.Entity
+        )
+
+        assertThat(capturedBob.entity.data["name"]).isEqualTo("bob")
+        assertThat(capturedBob.entity.data["age"]).isEqualTo(42)
```
Comment (csilvestrini) [inline javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: Optional: if this is the full set of data, a nicer way of testing this is:

```
assertThat(capturedBob.entity.data).containsExactly(
  "name", "bob",
  "age", 42
)
```

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+        activeStore.onProxyMessage(
+            ProxyMessage.ModelUpdate(RefModeStoreData.Set(collection.data), 1)
+        )
+
+        // Clone
+        val activeStore2 = createReferenceModeStore()
+        activeStore2.cloneFrom(activeStore)
+
+        assertThat(activeStore2.getLocalData()).isEqualTo(activeStore.getLocalData())
+        assertThat(activeStore2.getLocalData() === activeStore.getLocalData()).isFalse()
```
Comment (csilvestrini) [inline javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: isNotSameInstanceAs

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+
+        assertThat(capturedPeople.values).isEqualTo(referenceCollection.consumerView)
+        val storedBob = activeStore.backingStore.getLocalData("an-id") as CrdtEntity.Data
+        // Check that the stored bob's singleton data is equal to the expected bob's singleton data
+        assertThat(storedBob.singletons.mapValues { it.value.data })
+            .isEqualTo(bobEntity.data.singletons.mapValues { it.value.data })
+        // Check that the stored bob's collection data is equal to the expected bob's collection
+        // data (empty)
+        assertThat(storedBob.collections.mapValues { it.value.data })
+            .isEqualTo(bobEntity.data.collections.mapValues { it.value.data })
```
Comment (csilvestrini) [inline javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: I think you should just be able to do an equality check:

```
assertThat(storedBob.singletons).isEqualTo(bobEntity.data.singletons)
```

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt (hunk 2 above)
Comment (jasonwyatt) [reply javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: Ah that's what I get for copy-pasting the bulk of this 🙃 I'll fix it in the main RefModeStoreIntegrationTest as well.

File: javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt (hunk 3 above)
Comment (jasonwyatt) [reply javatests/arcs/core/storage/ReferenceModeStoreDatabaseIntegrationTest.kt]: Thanks!

**Pull request merged**